gio/src/application.rs

```diff
@@ -26,16 +26,17 @@ pub trait ApplicationExtManual {
 }
 
 impl<O: IsA<Application>> ApplicationExtManual for O {
-    fn run(&self) -> i32 {
+    fn run(&self) -> ExitCode {
         self.run_with_args(&std::env::args().collect::<Vec<_>>())
     }
 
-    fn run_with_args<S: AsRef<str>>(&self, args: &[S]) -> i32 {
+    fn run_with_args<S: AsRef<str>>(&self, args: &[S]) -> ExitCode {
         let argv: Vec<&str> = args.iter().map(|a| a.as_ref()).collect();
         let argc = argv.len() as i32;
-        unsafe {
+        let exit_code = unsafe {
             ffi::g_application_run(self.as_ref().to_glib_none().0, argc, argv.to_glib_none().0)
-        }
+        };
+        ExitCode::from(exit_code as u8)
```

Comment:
Isn't it common to return `-1`? That would not work anymore now.

Please check what `g_application_run()` can potentially return and if we would run into problems here.

Reply:
Yes but is that going to behave correctly? I guess so, as exit code on POSIX at least are always u8 anyway IIRC.

Where does this return value come from exactly, and can an application override what `run()` is returning somehow?

Reply:
> Where does this return value come from exactly, and can an application override what run() is returning somehow?

There are 
```
    fn command_line(&self, command_line: &crate::ApplicationCommandLine) -> i32 {
        self.parent_command_line(command_line)
    }

    fn local_command_line(&self, arguments: &mut ArgumentList) -> Option<i32> {
        self.parent_local_command_line(arguments)
    }
```
in ApplicationImpl, but the return type cannot be ExitCode as you cannot cast it back to u8/i32 and use that in the C trampoline.

> Yes but is that going to behave correctly? I guess so, as exit code on POSIX at least are always u8 anyway IIRC.

I have no clue, but ExitCode intentionally has only From<u8> for compatibility reasons.

Reply:
I think we can't really use `ExitCode` here then as that would break applications that return some specific `i32` there

Reply:
Or what do you think @A6GibKm ?

Reply:
I am not sure, losing info when doing i32 -> u8 is not ideal but tolerable and makes things portable... on the other hand not being able to read the u8... that might completely ruin someone's code, after app.run() might not be the actual exit code of an app, maybe https://doc.rust-lang.org/std/process/struct.ExitStatus.html its more appropiate? If not we can always do a wrapper around the i32 that implements the termination trait.

Reply:
I think implementing `Termination` around some wrapper type that allows getting back the original `i32` would make most sense.

Reply:
Added that